plugin/proxy/grpc_test.go

```diff
@@ -14,26 +14,51 @@ import (
 	"google.golang.org/grpc/grpclog"
 )
 
-func pool() []*healthcheck.UpstreamHost {
-	return []*healthcheck.UpstreamHost{
-		{
-			Name: "localhost:10053",
-		},
-		{
-			Name: "localhost:10054",
-		},
-	}
+func init() {
+	grpclog.SetLoggerV2(discardV2{})
 }
 
-func TestStartupShutdown(t *testing.T) {
-	grpclog.SetLogger(discard{})
+func buildPool(size int) ([]*healthcheck.UpstreamHost, func(), error) {
+	ups := make([]*healthcheck.UpstreamHost, size)
+	srvs := []*dns.Server{}
+	errs := []error{}
+	for i := 0; i < size; i++ {
+		srv, addr, err := test.TCPServer("localhost:0")
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		ups[i] = &healthcheck.UpstreamHost{Name: addr}
+		srvs = append(srvs, srv)
+	}
+	stopIt := func() {
+		for _, s := range srvs {
+			s.Shutdown()
+		}
+	}
+	if len(errs) > 0 {
+		go stopIt()
```

Comment:
If at lest one error when started the Servers, then stop all and return an error